WpAssets.php

```diff
@@ -169,14 +169,23 @@ private function findSrc($handle, $ext = '.js', $sep = '.') {
   }
 
   /**
-   * Uses the script_loader_tag filter to add crossorigin="anonymous" attribute to a specific script.
+   * Uses the script_loader_tag filter to add attributes to a specific script.
+   * For example to add crossorigin="use-credentials" to a script.
+   *
+   * If the value of the the attribute is a boolean, only the attribute will be
+   * set for example if async: true only async will be added to the script.
    *
    * @param  String  $name  The name of the script.
+   * @param  String  $attr  The name of the attribute ex: crossorigin.
+   * @param  String or Boolean $value The value of the attribute ex: "anonymous"
    */
-  public function addCrossoriginAttr($name) {
-    add_filter('script_loader_tag', function ($tag, $handle) use ($name) {
+  public function addAttr($name, $attr, $value) {
+    add_filter('script_loader_tag', function ($tag, $handle) use ($name, $attr, $value) {
       if ($name === $handle) {
-        return preg_replace('/<script( )*/', '<script crossorigin="anonymous"$1', $tag);
+        $script = '<script '.(is_bool($value) ? "$attr " : "$attr=$value" ).'$1';
```

Comment:
Just add a space around the periods for style. It looks like `$attr=$value`needs a space at the end of the string as well;

`'<script ' . (is_bool($value) ? "$attr " : "$attr=$value " ) . '$1';`